proto/redpanda/api/dataplane/v1alpha1/topic.proto

```diff
@@ -184,17 +184,17 @@ message UpdateTopicConfigurationsResponse {
   repeated Topic.Configuration configurations = 1;
 }
 
-message SetTopicConfigurationRequest {
+message SetTopicConfigurationsRequest {
   message SetConfiguration {
     string key = 1;
-    string value = 2;
+    optional string value = 2;
```

Comment:
This is whaat the Kafka client / Kafka API expects as well:

```go
type AlterConfigsRequestResourceConfig struct {
	// Name is a key to set (e.g. segment.bytes).
	Name string

	// Value is a value to set for the key (e.g. 10).
	Value *string
}
```

This could be to set a certain config property to "nil" at the topic config source. In Apache Kafka for example this setting exists `follower.replication.throttled.replicas` you may set this at the broker/cluster level but can override this at the topic level by setting it to 0. There may be more configs where this is useful today or in the future.